pallets/loans/src/entities/pricing/external.rs

```diff
@@ -94,36 +125,44 @@ impl<T: Config> ExternalActivePricing<T> {
 	) -> Result<T::Balance, DispatchError> {
 		match self.info.max_borrow_amount {
 			MaxBorrowAmount::Quantity(quantity) => {
-				let price = self.calculate_price()?;
+				let price = self.current_price()?;
 				let available = quantity.ensure_sub(self.outstanding_quantity)?;
 				Ok(price.ensure_mul_int(available)?)
 			}
 			MaxBorrowAmount::NoLimit => Ok(desired_amount),
 		}
 	}
 
-	pub fn last_updated(&self) -> Result<Moment, DispatchError> {
-		Ok(T::PriceRegistry::get(&self.info.price_id)?.1)
-	}
+	pub fn adjust(
+		&mut self,
+		principal_adj: Adjustment<T::Balance>,
+		interest: T::Balance,
+	) -> DispatchResult {
+		let quantity_adj = principal_adj.try_map(|principal| -> Result<_, DispatchError> {
+			let price = self.current_price()?;
 
-	pub fn compute_present_value(&self, price: T::Rate) -> Result<T::Balance, DispatchError> {
-		Ok(price.ensure_mul_int(self.outstanding_quantity)?)
-	}
+			let quantity = T::Rate::saturating_from_integer(principal)
+				.ensure_div(price)?
+				.ensure_mul_int(One::one())?;
+
+			ensure!(
+				price.ensure_mul_int(quantity)? == principal,
+				Error::<T>::AmountNotMultipleOfPrice
+			);
+
+			Ok(quantity)
+		})?;
 
-	pub fn adjust_debt(&mut self, adjustment: Adjustment<T::Balance>) -> DispatchResult {
-		let price = self.calculate_price()?;
-		let amount = adjustment.abs();
-		let quantity = T::Rate::one().ensure_div(price)?.ensure_mul_int(amount)?;
+		self.outstanding_quantity = quantity_adj.ensure_add(self.outstanding_quantity)?;
 
-		ensure!(
-			price.ensure_mul_int(quantity)? == amount,
-			Error::<T>::AmountNotMultipleOfPrice
-		);
+		let interest_adj = quantity_adj.try_map(|quantity| {
+			self.info
+				.notional
+				.ensure_mul_int(quantity)?
+				.ensure_add(interest)
+		})?;
```

Comment:
Why must the notinal be multiplied by the `quantity`? Is the notional always the value of `1` asset? 

Could you maybe explain that a bit? I.e. how the interest adjustment works?

Reply:
>  Is the notional always the value of 1 asset?

Yeah exactly, notional value is the value of 1 asset (e.g. 1 T-bill). This can basically be considered the "reference amount that interest accrues on". This notional value never changes, only the price (valuation) does.

So the interest should accrue on `notional x outstanding quantity`

Reply:
The notional is the price used as a reference to calculate interest. Here's the [slack thread](https://centrifugedao.slack.com/archives/C05DX0RH1T3/p1687857589371469) with the @offerijns explanation about this:

Reply:
But why do we need the multiplication? Shouldn't the `interest` simply be added?

Reply:
Because the notional is the price of only one treasury, you can have `quantity` treasuries